components/zcash_address/src/kind/unified/fvk.rs

```diff
@@ -0,0 +1,360 @@
+use std::cmp;
+use std::convert::{TryFrom, TryInto};
+
+use super::{
+    private::{SealedContainer, SealedItem},
+    Container, Encoding, ParseError, Typecode,
+};
+
+/// The set of known FVKs for Unified FVKs.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Fvk {
+    /// The raw encoding of an Orchard Full Viewing Key.
+    ///
+    /// `(ak, nk, rivk)` each 32 bytes.
+    Orchard([u8; 96]),
+
+    /// Data contained within the Sapling component of a Unified Full Viewing Key
+    ///
+    /// `(ak, nk, ovk, dk)` each 32 bytes.
+    Sapling([u8; 128]),
+
+    /// A pruned version of the extended public key for the BIP 44 account corresponding to the
+    /// transparent address subtree from which transparent addresses are derived. This
+    /// includes just the chain code (32 bytes) and the public key (33 bytes) and excludes
+    /// the depth of in the derivation tree, the parent key fingerprint, and the child key
+    /// number (which would reveal the wallet account number for which this UFVK was generated).
+    ///
+    /// Transparent addresses don't have "viewing keys" - the addresses themselves serve
+    /// that purpose. However, we want the ability to derive diversified Unified Addresses
+    /// from Unified Viewing Keys, and to not break the unlinkability property when they
+    /// include transparent receivers. To achieve this, we treat the last hardened node in
+    /// the BIP 44 derivation path as the "transparent viewing key"; all addresses derived
+    /// from this node use non-hardened derivation, and can thus be derived just from this
+    /// extended public key.
+    P2pkh([u8; 65]),
+
+    Unknown {
+        typecode: u32,
+        data: Vec<u8>,
+    },
+}
+
+impl cmp::Ord for Fvk {
```

Comment:
I wonder if we can just have a common `impl<T: SealedItem> cmp::Ord for T` (and similarly for `PartialOrd`)?